src/server/journal/streamer.cc

```diff
@@ -215,6 +219,9 @@ void RestoreStreamer::Run() {
       return;
 
     cursor = db_slice_->Traverse(pt, cursor, [&](PrimeTable::bucket_iterator it) {
+      if (fiber_cancelled_)  // Could be cancelled any time as Traverse may preempt
```

Comment:
how can traverse preempt if we dont have the big value serialization merged yet?

Reply:
in the callback

Reply:
which callback can preempt?

Reply:
by the way I think we can use snapshot_version_ instead of fiber_canceled_ because we always process them together

Reply:
> which callback can preempt?

db_slice_->FlushChangeToEarlierCallbacks(0 /*db_id always 0 for cluster*/,
                                               DbSlice::Iterator::FromPrime(it), snapshot_version_);

Reply:
also WriteBucket(it); can yield

Reply:
so I think the if fiber_cancelled_ should be also after call to FlushChangeToEarlierCallbacks

Reply:
why?